src/modules/launcher/Microsoft.Launcher/dllmain.cpp

```diff
@@ -93,11 +104,10 @@ class Microsoft_Launcher : public PowertoyModuleIface
         Logger::info("Launcher object is constructing");
         init_settings();
 
-        SECURITY_ATTRIBUTES sa;
-        sa.nLength = sizeof(sa);
-        sa.bInheritHandle = false;
-        sa.lpSecurityDescriptor = NULL;
-        m_hEvent = CreateEventW(&sa, FALSE, FALSE, CommonSharedConstants::POWER_LAUNCHER_SHARED_EVENT);
+        auto sa1 = getDefaultSecurityAttribute();
+        m_hEvent = CreateEventW(&sa1, FALSE, FALSE, CommonSharedConstants::POWER_LAUNCHER_SHARED_EVENT);
+        auto sa2 = getDefaultSecurityAttribute();
+        send_telemetry_event = CreateEventW(&sa2, FALSE, FALSE, CommonSharedConstants::SEND_RUN_SETTINGS_TELEMETRY_EVENT);
```

Comment:
It would be useful to have an helper function in common to be reused for other modules as well.
But we don't need to do it now, let's open an issue to keep track of this work.

Reply:
A function that by event name will create an auto-reset event, right?